tough/src/schema/mod.rs

```diff
@@ -224,6 +228,45 @@ pub struct Target {
     pub _extra: HashMap<String, Value>,
 }
 
+impl Target {
+    /// Given a path, returns a Target struct
+    pub fn from_path<P>(path: P) -> Result<Target>
```

Comment:
I should mention that I initially tried to implement `TryFrom<AsRef<Path>>` and ran into [this Rust issue](https://github.com/rust-lang/rust/issues/50133). After that, I didn't think this was worth it to implement `TryFrom` with `String`, `Path`, `PathBuf`, etc.